### PR TITLE
98192 Update alert on choose a VA location page - Part 9

### PR DESCRIPTION
**Pull request by vbahinwillit**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR updates the warning message content.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#98192

## Testing done
Unit and e2e

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |  localhost_3001_my-health_appointments_schedule_location (7)      |   localhost_3001_my-health_appointments_schedule_location (9)    |

## What areas of the site does it impact?
VAOS

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx [hunk 1]
```diff
@@ -20,18 +20,24 @@ export default function EligibilityModal({
   return (
     <VaModal
       id="eligibilityModal"
-      status="warning"
+      status={status}
       visible
       onCloseEvent={onClose}
-      hideCloseButton={status === FETCH_STATUS.loading}
```
Comment (JunTaoLuo) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:26]: Hmm why was this removed?

File: src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx [hunk 2]
```diff
@@ -20,18 +20,24 @@ export default function EligibilityModal({
...
       modalTitle={title}
       ariaLabel={title}
       data-testid="eligibilityModal"
       role="alertdialog"
       uswds
     >
       <div aria-atomic="true" aria-live="assertive">
-        {content}
+        <p>{content}</p>
+        <p>
```
Comment (JunTaoLuo) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:34]: Is it safe to add this content in this component? It would make sense if this modal is only used for 502 errors or if all eligibility messages share this content. Going off of the Figma file from design I'm having trouble parsing if either of these cases are true so can you confirm why we can add this content here instead of `getEligibilityMessage`?

**Review by JunTaoLuo — COMMENTED** [review]
Changes look good but had a couple of questions.

File: src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:34]: @JunTaoLuo - Added here due to not being able to add html formatting to the content string. The html formatting displays as part of the string. I did check and the EligibilityModal is only used by the 'VAFacilityPageV2' component right now. The added message seems generic enough to apply to any error message.

File: src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:26]: @JunTaoLuo - Not used.

File: src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx (hunk 2 above)
Comment (JunTaoLuo) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:34]: Hmm if we are adding something to the error message, shouldn't we run that by the design folks? I see that it is only used by `VAFacilityPageV2` but what I'm concerned about is if it is used to display other types of errors, some of which shouldn't have the additional content.

Comment (JunTaoLuo) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:34]: Hmm so I tried the scenario where we facility eligibility fails due to `facility-request-limit-exceeded` and I see the new content appended to the end of the warning:
image
Compared to what's in the Figma files:
image
Can we just make the new content conditional on the `status` variable and only display it if it is `== 'error'`?

Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/EligibilityModal.jsx:34]: Yes. Thanks.

**Review by JunTaoLuo — APPROVED** [review]
LGTM!
